python/ray/ml/examples/pytorch/off_the_shelf_fashion_mnist.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+
+from torch import nn, optim
+from torchvision import datasets, transforms
+
+import ray
+from ray.ml.train.integrations.torch.torch_ots import TorchOffTheShelfTrainer
+
+
+def get_dataset(train: bool = False) -> ray.data.Dataset:
+    """Download FashionMNIST dataset and return as Ray dataset."""
+    training_data = datasets.FashionMNIST(
+        root="~/data",
+        train=train,
+        download=True,
+        transform=transforms.Lambda(lambda img: np.array(img).tolist()),
+    )
+    ds = ray.data.from_items([{"x": x, "y": y} for x, y in training_data])
+    return ds
+
+
+class NeuralNetwork(nn.Module):
+    """Simple MLP with three layers."""
+
+    def __init__(self):
+        super(NeuralNetwork, self).__init__()
+        self.flatten = nn.Flatten()
+        self.linear_relu_stack = nn.Sequential(
+            nn.Linear(28 * 28, 512),
+            nn.ReLU(),
+            nn.Linear(512, 512),
+            nn.ReLU(),
+            nn.Linear(512, 10),
+            nn.ReLU(),
+        )
+
+    def forward(self, x):
+        x = self.flatten(x)
+        return self.linear_relu_stack(x)
+
+
+# Initialize the off-the-shelf trainer and fit to the dataset
+
+trainer = TorchOffTheShelfTrainer(
```

Comment:
@krfricke thanks for this RFC! 

I think you could argue that something like this (not 100% about the exact specs yet) should be the default trainer users see. I'm not sure how often you really want to implement your own train loop in practice. This abstraction is certainly quicker to the punch for many use cases.

As you can imagine, I don't like the name. :D An idea would be to call this one `TorchTrainer` and the more complex underlying one something else.

Having said all that, I want to gauge the overall feeling about this from the rest of the pack first. As this diverges from the original design, I'm not sure we want to introduce another layer right now.

Reply:
Thanks for the feedback. Yes, this would be a higher level layer to make it very easy for users to start off. If we decide to do something like this, we should try to collect a lot of feedback to be sure we provide the right amount of customization while still keeping it very, very simple.

I'm always open to better naming ;-)